### ISSUE TITLE
[bug?] parsing styles with external composes rule

### ISSUE THREAD

**Issue by kubijo** [skovhus/css-modules-flow-types#5]
Hi Kenneth.

First and foremost thanks for your efforts.
I have however trouble setting up the module in my project. (edit: I'm using the webpack loader)

My proble is that css-module classname that use `composes` rule from different stylesheet results in the intermediate source ilustrated in the pseudo code below.

Here is the code generating loader config for webpack:
![selection_001](https://user-images.githubusercontent.com/11244314/31494552-1038c280-af54-11e7-8d94-3da9f81d06ba.png)

**TLDR:**
```javascript
ExtractTextPlugin.extract({
    fallback: 'style-loader',
    use: [
        'css-modules-flow-types'
        'css-loader',
        'postcss-loader',
    ]
});
```

And this is the situation in stylesheet vs transpiled code that you are trying to parse in the webpack loader:

```css
/* Menu.scss */
.root {
    composes: foo from 'styles/bar.scss';
    color: red;
}
```

will result in something like this source passed to your loader (in a string form)

```javascript
{
    "root": "Menu_root_1TJ" + require("-!css-loader!styles/bar.scss").locals["foo"] + "",
}
```

Now this obviously causes [`JSON.parse()`](https://github.com/skovhus/css-modules-flow-types/blob/3e11c479a0a07dcf18fdeb89f5a362520569ce2d/packages/css-modules-flow-types-loader/index.js#L9) to fail as soon as the `+` token is reached.

---

Am I doing something wrong in the setup or is this something that was not accounted for?
As the source for the loader is pretty simple and concise, I'm ready to create a pull-request, but I'd like to settle on some commonly agreed solution.

My ideas if this a bug:
 - regex parsing of object keys as the only part you are interested in is the list of local classes
 - sanitation of the input source by clearing out `require` calls and concatenations (essentially the same as first one, but now the `JSON.parse` should be fine

**Comment by skovhus** [comment]
Thanks for this detailed report!

Can you try the `css-modules-flow-types-cli`? And see if that works for you. 

https://github.com/skovhus/css-modules-flow-types/tree/master/packages/css-modules-flow-types-cli

It uses a different approach than doing a `JSON.parse` (like the webpack loader). 

> Am I doing something wrong in the setup or is this something that was not accounted for?

I didn't think of this case. It is clearly something that needs a fix.

> I'm ready to create a pull-request

That would be awesome! I'll happily help you. If I were you I would start by documenting the bug with a unit test (https://github.com/skovhus/css-modules-flow-types/blob/master/packages/css-modules-flow-types-loader/__test__/index.test.js). And then figure out what is the best solution to the problem.

The cleanest might be to update the regex here https://github.com/skovhus/css-modules-flow-types/blob/master/packages/css-modules-flow-types-loader/index.js#L7

Again, thanks for wanting to contribute!

**Comment by kubijo** [comment]
The CLI package does not seem do to anything for me ... but my stylesheets use quite an elaborate pre-processing anyway (pre-css, paths resolving by webpack, etc…)

I'll go for the test + regex path and test it ... if you could point me to a quick solution for using the edited version of loader, that would be a good kick in the right direction…

**Comment by skovhus** [comment]
> if you could point me to a quick solution for using the edited version of loader, that would be a good kick in the right direction…

You mean using a local version of `css-modules-flow-types-loader`?

**Comment by kubijo** [comment]
Yes, haven't worked on that before ...

**Comment by skovhus** [comment]
Sure. So after [forking and cloning](https://help.github.com/articles/fork-a-repo/) and then you go to your folder with `css-modules-flow-types/packages/css-modules-flow-types-loader` and do

     yarn link  # or npm link
 
And go to the root of your project where you want to test it and run 

    yarn link css-modules-flow-types-loader


See more at https://yarnpkg.com/lang/en/docs/cli/link/

**Comment by kubijo** [comment]
Cool … will post update :+1:

**Comment by skovhus** [comment]
But I'll like to know what goes on with the CLI package and your code. It should at least fail if it cannot create the files. : )

**Comment by kubijo** [comment]
I'll try again once this is sorted out.

One question regarding the printer though:
You are currently passing tokens as an object, but using just the keys.
Do you have anything against changing that to just an array of tokens (classNames)?

**Comment by skovhus** [comment]
Released as 1.0.1. Thank you for contributing! : )
